{"uid":"236e5a5a5754c13f","name":"test_rpc_node_is_synced[Test for Khala, url: wss://khala-api.phala.network/ws]","fullName":"tests.test_substrate_node_is_synced#test_rpc_node_is_synced","historyId":"30247941ab5ba60a864b6648c50384f3","time":{"start":1731683144349,"stop":1731683144349,"duration":0},"status":"broken","statusMessage":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:45 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=Et1vKjW5hyko3wA5SFW7KS4zTuUiwethkvQUuW1Sy%2BOtbS2zPgBC9BI8WNcDUqF14iOUgThYQLkku58K0SzDauNzHCB4Jha0w6%2Fa%2F1y9xWsUBdnI9YsPgwI3MzWSwIB06CfqZz%2BAVJd95A%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e302425da9a7d89-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8496&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2829&recv_bytes=824&delivery_rate=330440&cwnd=250&unsent_bytes=0&cid=447ced45f7b75598&ts=461&x=0\"'} -+-+- b'error code: 502'","statusTrace":"request = <SubRequest 'connection_by_url' for <Function test_rpc_node_is_synced[Test for Khala, url: wss://khala-api.phala.network/ws]>>\n\n    @pytest.fixture(scope=\"module\", params=collect_nodes_for_chains(get_substrate_chains()), ids=task_ids)\n    def connection_by_url(request):\n        data = request.param\n>       connection = execute_with_timeout(\n            timeout=FIXTURE_TIMEOUT,\n            function=create_connection_by_url,\n            args=(data[\"url\"],),\n            error_message=f'Timeout {FIXTURE_TIMEOUT} when creating connection: {data[\"url\"]}'\n        )\n\ntests/conftest.py:57: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ntests/conftest.py:32: in execute_with_timeout\n    return func_timeout(timeout=timeout, func=function, args=args)\n.venv/lib/python3.10/site-packages/func_timeout/dafunc.py:108: in func_timeout\n    raise_exception(exception)\n.venv/lib/python3.10/site-packages/func_timeout/py3_raise.py:7: in raise_exception\n    raise exception[0] from None\nscripts/utils/substrate_interface.py:30: in create_connection_by_url\n    substrate = SubstrateInterface(\n.venv/lib/python3.10/site-packages/substrateinterface/base.py:132: in __init__\n    self.connect_websocket()\n.venv/lib/python3.10/site-packages/substrateinterface/base.py:177: in connect_websocket\n    self.websocket = create_connection(\n.venv/lib/python3.10/site-packages/websocket/_core.py:646: in create_connection\n    websock.connect(url, **options)\n.venv/lib/python3.10/site-packages/websocket/_core.py:261: in connect\n    self.handshake_response = handshake(self.sock, url, *addrs, **options)\n.venv/lib/python3.10/site-packages/websocket/_handshake.py:65: in handshake\n    status, resp = _get_resp_headers(sock)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nsock = <ssl.SSLSocket [closed] fd=-1, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6>\nsuccess_statuses = (<HTTPStatus.MOVED_PERMANENTLY: 301>, <HTTPStatus.FOUND: 302>, <HTTPStatus.SEE_OTHER: 303>, <HTTPStatus.TEMPORARY_REDIRECT: 307>, <HTTPStatus.PERMANENT_REDIRECT: 308>, <HTTPStatus.SWITCHING_PROTOCOLS: 101>)\n\n    def _get_resp_headers(sock, success_statuses: tuple = SUCCESS_STATUSES) -> tuple:\n        status, resp_headers, status_message = read_headers(sock)\n        if status not in success_statuses:\n            content_len = resp_headers.get(\"content-length\")\n            if content_len:\n                response_body = sock.recv(\n                    int(content_len)\n                )  # read the body of the HTTP error message response and include it in the exception\n            else:\n                response_body = None\n>           raise WebSocketBadStatusException(\n                f\"Handshake status {status} {status_message} -+-+- {resp_headers} -+-+- {response_body}\",\n                status,\n                status_message,\n                resp_headers,\n                response_body,\n            )\nE           websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:45 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=Et1vKjW5hyko3wA5SFW7KS4zTuUiwethkvQUuW1Sy%2BOtbS2zPgBC9BI8WNcDUqF14iOUgThYQLkku58K0SzDauNzHCB4Jha0w6%2Fa%2F1y9xWsUBdnI9YsPgwI3MzWSwIB06CfqZz%2BAVJd95A%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e302425da9a7d89-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8496&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2829&recv_bytes=824&delivery_rate=330440&cwnd=250&unsent_bytes=0&cid=447ced45f7b75598&ts=461&x=0\"'} -+-+- b'error code: 502'\n\n.venv/lib/python3.10/site-packages/websocket/_handshake.py:150: WebSocketBadStatusException","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":2,"retriesStatusChange":false,"beforeStages":[{"name":"connection_by_url","time":{"start":1731683132711,"stop":1731683133446,"duration":735},"status":"broken","statusMessage":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:33 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=7wgOJp3xKJnjpGVg47DHvjkz7Xde6vlyU30xHAd3HEKPc8bpNHqZhFF1K5bCo8NPyVB2yhUxsA7vmCv5RpQugMZ%2BBdrYAOKM8JmXrDtB16clOxwgedGkmHOYP%2Fh4l159fuH8CjHoHB3t9A%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e3023dd1e58315b-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8398&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2828&recv_bytes=824&delivery_rate=344760&cwnd=248&unsent_bytes=0&cid=102a19ab2b92eef3&ts=489&x=0\"'} -+-+- b'error code: 502'\n","statusTrace":"  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1128, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 907, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/tests/conftest.py\", line 57, in connection_by_url\n    connection = execute_with_timeout(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/tests/conftest.py\", line 32, in execute_with_timeout\n    return func_timeout(timeout=timeout, func=function, args=args)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/func_timeout/dafunc.py\", line 108, in func_timeout\n    raise_exception(exception)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/func_timeout/py3_raise.py\", line 7, in raise_exception\n    raise exception[0] from None\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/scripts/utils/substrate_interface.py\", line 30, in create_connection_by_url\n    substrate = SubstrateInterface(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/substrateinterface/base.py\", line 132, in __init__\n    self.connect_websocket()\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/substrateinterface/base.py\", line 177, in connect_websocket\n    self.websocket = create_connection(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_core.py\", line 646, in create_connection\n    websock.connect(url, **options)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_core.py\", line 261, in connect\n    self.handshake_response = handshake(self.sock, url, *addrs, **options)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_handshake.py\", line 65, in handshake\n    status, resp = _get_resp_headers(sock)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_handshake.py\", line 150, in _get_resp_headers\n    raise WebSocketBadStatusException(\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"connection_by_url","time":{"start":1731683138568,"stop":1731683139284,"duration":716},"status":"broken","statusMessage":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:39 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=drTIxD3jHWT3LXJPXKJ1fmcO0JGyjiZ2LeUF8zATsltnFrRz0ScR48KTSjtIQItFduez%2BUHAJ%2BUsTLKTFfmDnO40RgX06Ubry0Ot6KHrV2W35VRvahSYsdBydEBu1xUx87i2YusB%2F7iy4Q%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e302401b90308d1-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8470&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2829&recv_bytes=824&delivery_rate=334975&cwnd=233&unsent_bytes=0&cid=aaf3f58d52974365&ts=456&x=0\"'} -+-+- b'error code: 502'\n","statusTrace":"  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1128, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 907, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/tests/conftest.py\", line 57, in connection_by_url\n    connection = execute_with_timeout(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/tests/conftest.py\", line 32, in execute_with_timeout\n    return func_timeout(timeout=timeout, func=function, args=args)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/func_timeout/dafunc.py\", line 108, in func_timeout\n    raise_exception(exception)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/func_timeout/py3_raise.py\", line 7, in raise_exception\n    raise exception[0] from None\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/scripts/utils/substrate_interface.py\", line 30, in create_connection_by_url\n    substrate = SubstrateInterface(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/substrateinterface/base.py\", line 132, in __init__\n    self.connect_websocket()\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/substrateinterface/base.py\", line 177, in connect_websocket\n    self.websocket = create_connection(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_core.py\", line 646, in create_connection\n    websock.connect(url, **options)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_core.py\", line 261, in connect\n    self.handshake_response = handshake(self.sock, url, *addrs, **options)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_handshake.py\", line 65, in handshake\n    status, resp = _get_resp_headers(sock)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_handshake.py\", line 150, in _get_resp_headers\n    raise WebSocketBadStatusException(\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"connection_by_url","time":{"start":1731683144349,"stop":1731683145065,"duration":716},"status":"broken","statusMessage":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:45 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=Et1vKjW5hyko3wA5SFW7KS4zTuUiwethkvQUuW1Sy%2BOtbS2zPgBC9BI8WNcDUqF14iOUgThYQLkku58K0SzDauNzHCB4Jha0w6%2Fa%2F1y9xWsUBdnI9YsPgwI3MzWSwIB06CfqZz%2BAVJd95A%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e302425da9a7d89-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8496&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2829&recv_bytes=824&delivery_rate=330440&cwnd=250&unsent_bytes=0&cid=447ced45f7b75598&ts=461&x=0\"'} -+-+- b'error code: 502'\n","statusTrace":"  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/pluggy/_callers.py\", line 77, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1128, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/_pytest/fixtures.py\", line 907, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/tests/conftest.py\", line 57, in connection_by_url\n    connection = execute_with_timeout(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/tests/conftest.py\", line 32, in execute_with_timeout\n    return func_timeout(timeout=timeout, func=function, args=args)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/func_timeout/dafunc.py\", line 108, in func_timeout\n    raise_exception(exception)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/func_timeout/py3_raise.py\", line 7, in raise_exception\n    raise exception[0] from None\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/scripts/utils/substrate_interface.py\", line 30, in create_connection_by_url\n    substrate = SubstrateInterface(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/substrateinterface/base.py\", line 132, in __init__\n    self.connect_websocket()\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/substrateinterface/base.py\", line 177, in connect_websocket\n    self.websocket = create_connection(\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_core.py\", line 646, in create_connection\n    websock.connect(url, **options)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_core.py\", line 261, in connect\n    self.handshake_response = handshake(self.sock, url, *addrs, **options)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_handshake.py\", line 65, in handshake\n    status, resp = _get_resp_headers(sock)\n  File \"/home/runner/work/test-runner/test-runner/nova-utils/.venv/lib/python3.10/site-packages/websocket/_handshake.py\", line 150, in _get_resp_headers\n    raise WebSocketBadStatusException(\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"testStage":{"status":"broken","statusMessage":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:45 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=Et1vKjW5hyko3wA5SFW7KS4zTuUiwethkvQUuW1Sy%2BOtbS2zPgBC9BI8WNcDUqF14iOUgThYQLkku58K0SzDauNzHCB4Jha0w6%2Fa%2F1y9xWsUBdnI9YsPgwI3MzWSwIB06CfqZz%2BAVJd95A%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e302425da9a7d89-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8496&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2829&recv_bytes=824&delivery_rate=330440&cwnd=250&unsent_bytes=0&cid=447ced45f7b75598&ts=461&x=0\"'} -+-+- b'error code: 502'","statusTrace":"request = <SubRequest 'connection_by_url' for <Function test_rpc_node_is_synced[Test for Khala, url: wss://khala-api.phala.network/ws]>>\n\n    @pytest.fixture(scope=\"module\", params=collect_nodes_for_chains(get_substrate_chains()), ids=task_ids)\n    def connection_by_url(request):\n        data = request.param\n>       connection = execute_with_timeout(\n            timeout=FIXTURE_TIMEOUT,\n            function=create_connection_by_url,\n            args=(data[\"url\"],),\n            error_message=f'Timeout {FIXTURE_TIMEOUT} when creating connection: {data[\"url\"]}'\n        )\n\ntests/conftest.py:57: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ntests/conftest.py:32: in execute_with_timeout\n    return func_timeout(timeout=timeout, func=function, args=args)\n.venv/lib/python3.10/site-packages/func_timeout/dafunc.py:108: in func_timeout\n    raise_exception(exception)\n.venv/lib/python3.10/site-packages/func_timeout/py3_raise.py:7: in raise_exception\n    raise exception[0] from None\nscripts/utils/substrate_interface.py:30: in create_connection_by_url\n    substrate = SubstrateInterface(\n.venv/lib/python3.10/site-packages/substrateinterface/base.py:132: in __init__\n    self.connect_websocket()\n.venv/lib/python3.10/site-packages/substrateinterface/base.py:177: in connect_websocket\n    self.websocket = create_connection(\n.venv/lib/python3.10/site-packages/websocket/_core.py:646: in create_connection\n    websock.connect(url, **options)\n.venv/lib/python3.10/site-packages/websocket/_core.py:261: in connect\n    self.handshake_response = handshake(self.sock, url, *addrs, **options)\n.venv/lib/python3.10/site-packages/websocket/_handshake.py:65: in handshake\n    status, resp = _get_resp_headers(sock)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nsock = <ssl.SSLSocket [closed] fd=-1, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6>\nsuccess_statuses = (<HTTPStatus.MOVED_PERMANENTLY: 301>, <HTTPStatus.FOUND: 302>, <HTTPStatus.SEE_OTHER: 303>, <HTTPStatus.TEMPORARY_REDIRECT: 307>, <HTTPStatus.PERMANENT_REDIRECT: 308>, <HTTPStatus.SWITCHING_PROTOCOLS: 101>)\n\n    def _get_resp_headers(sock, success_statuses: tuple = SUCCESS_STATUSES) -> tuple:\n        status, resp_headers, status_message = read_headers(sock)\n        if status not in success_statuses:\n            content_len = resp_headers.get(\"content-length\")\n            if content_len:\n                response_body = sock.recv(\n                    int(content_len)\n                )  # read the body of the HTTP error message response and include it in the exception\n            else:\n                response_body = None\n>           raise WebSocketBadStatusException(\n                f\"Handshake status {status} {status_message} -+-+- {resp_headers} -+-+- {response_body}\",\n                status,\n                status_message,\n                resp_headers,\n                response_body,\n            )\nE           websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:45 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=Et1vKjW5hyko3wA5SFW7KS4zTuUiwethkvQUuW1Sy%2BOtbS2zPgBC9BI8WNcDUqF14iOUgThYQLkku58K0SzDauNzHCB4Jha0w6%2Fa%2F1y9xWsUBdnI9YsPgwI3MzWSwIB06CfqZz%2BAVJd95A%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e302425da9a7d89-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8496&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2829&recv_bytes=824&delivery_rate=330440&cwnd=250&unsent_bytes=0&cid=447ced45f7b75598&ts=461&x=0\"'} -+-+- b'error code: 502'\n\n.venv/lib/python3.10/site-packages/websocket/_handshake.py:150: WebSocketBadStatusException","steps":[],"attachments":[{"uid":"69ddc7799a2ca490","name":"stdout","source":"69ddc7799a2ca490.txt","type":"text/plain","size":3402}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[],"labels":[{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_substrate_node_is_synced"},{"name":"host","value":"fv-az1196-256"},{"name":"thread","value":"2123-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_substrate_node_is_synced"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"connection_by_url","value":"{'url': 'wss://khala-api.phala.network/ws', 'name': 'Khala'}"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[{"uid":"98b8e6402f4db90b","status":"broken","statusDetails":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:39 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=drTIxD3jHWT3LXJPXKJ1fmcO0JGyjiZ2LeUF8zATsltnFrRz0ScR48KTSjtIQItFduez%2BUHAJ%2BUsTLKTFfmDnO40RgX06Ubry0Ot6KHrV2W35VRvahSYsdBydEBu1xUx87i2YusB%2F7iy4Q%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e302401b90308d1-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8470&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2829&recv_bytes=824&delivery_rate=334975&cwnd=233&unsent_bytes=0&cid=aaf3f58d52974365&ts=456&x=0\"'} -+-+- b'error code: 502'","time":{"start":1731683138568,"stop":1731683138568,"duration":0}},{"uid":"c361a1d555ff89cd","status":"broken","statusDetails":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Fri, 15 Nov 2024 15:05:33 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=7wgOJp3xKJnjpGVg47DHvjkz7Xde6vlyU30xHAd3HEKPc8bpNHqZhFF1K5bCo8NPyVB2yhUxsA7vmCv5RpQugMZ%2BBdrYAOKM8JmXrDtB16clOxwgedGkmHOYP%2Fh4l159fuH8CjHoHB3t9A%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e3023dd1e58315b-LAX', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=8398&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2828&recv_bytes=824&delivery_rate=344760&cwnd=248&unsent_bytes=0&cid=102a19ab2b92eef3&ts=489&x=0\"'} -+-+- b'error code: 502'","time":{"start":1731683132710,"stop":1731683132710,"duration":0}}],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":5,"skipped":0,"passed":58,"unknown":0,"total":63},"items":[{"uid":"71da3fc169c3a0db","reportUrl":"https://novasamatech.github.io/test-runner/5832//#testresult/71da3fc169c3a0db","status":"passed","time":{"start":1731661579656,"stop":1731661579679,"duration":23}},{"uid":"c70fb36be5a2e42a","reportUrl":"https://novasamatech.github.io/test-runner/5831//#testresult/c70fb36be5a2e42a","status":"broken","statusDetails":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Thu, 14 Nov 2024 15:05:33 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=N0TfZrBZ%2FM2AgN13uG2XtDMlNd%2FzYsXOO%2FO7ZO9czRXXnsom3mrMbczDCKKnfriz7nDra48U2sSOPemTmfq8vLTvWijqwuOemQPCMIVpH7eDGJo4OPDNsfJXNVlCvL1L2pAbg97EsrgY9w%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e27e67fadd2d697-IAD', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=1334&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2828&recv_bytes=824&delivery_rate=2250591&cwnd=250&unsent_bytes=0&cid=b59a5327f0ca0739&ts=99&x=0\"'} -+-+- b'error code: 502'","time":{"start":1731596733120,"stop":1731596733120,"duration":0}},{"uid":"42c9cf19bd9e18b2","reportUrl":"https://novasamatech.github.io/test-runner/5830//#testresult/42c9cf19bd9e18b2","status":"passed","time":{"start":1731575149256,"stop":1731575149525,"duration":269}},{"uid":"6871e8133c266f55","reportUrl":"https://novasamatech.github.io/test-runner/5829//#testresult/6871e8133c266f55","status":"broken","statusDetails":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Wed, 13 Nov 2024 15:05:31 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=KFMdnXkdzev0i2YlionbBB8asdzx%2FkVtR5QGkak1BwGtY1zNx7JOv4k6JnGDDfJ7DVTgAMWezX7LLYg5CDIdKuHfNkeP202O16UUqYbGRca2hQpV4d%2FxjecUcPCvBdMZqOHqoM8ELdKhgQ%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e1fa91129a0a1e1-YYZ', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=13261&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2829&recv_bytes=824&delivery_rate=217782&cwnd=249&unsent_bytes=0&cid=0fe62f3705d80f4b&ts=299&x=0\"'} -+-+- b'error code: 502'","time":{"start":1731510330770,"stop":1731510330770,"duration":0}},{"uid":"10c21c3e4c397e7f","reportUrl":"https://novasamatech.github.io/test-runner/5828//#testresult/10c21c3e4c397e7f","status":"passed","time":{"start":1731488751929,"stop":1731488752008,"duration":79}},{"uid":"7d4dead4694ae5e4","reportUrl":"https://novasamatech.github.io/test-runner/5827//#testresult/7d4dead4694ae5e4","status":"passed","time":{"start":1731423935537,"stop":1731423935562,"duration":25}},{"uid":"918d37fb2a989caa","reportUrl":"https://novasamatech.github.io/test-runner/5826//#testresult/918d37fb2a989caa","status":"passed","time":{"start":1731402366212,"stop":1731402366429,"duration":217}},{"uid":"bed26a1807bc7bc7","reportUrl":"https://novasamatech.github.io/test-runner/5825//#testresult/bed26a1807bc7bc7","status":"passed","time":{"start":1731337528940,"stop":1731337529164,"duration":224}},{"uid":"df49ed5a14437ed4","reportUrl":"https://novasamatech.github.io/test-runner/5824//#testresult/df49ed5a14437ed4","status":"broken","statusDetails":"websocket._exceptions.WebSocketBadStatusException: Handshake status 502 Bad Gateway -+-+- {'date': 'Mon, 11 Nov 2024 09:06:03 GMT', 'content-type': 'text/plain; charset=UTF-8', 'content-length': '15', 'connection': 'keep-alive', 'report-to': '{\"endpoints\":[{\"url\":\"https:\\\\/\\\\/a.nel.cloudflare.com\\\\/report\\\\/v4?s=7GBVMOqSadRzay0zYX4Y%2FOFYMZFR6I5zCvGe1BUmbR6ftGCTVtpaQl%2FFSMxLVh8JM%2FTrcj3FQztRLqsOiGZ7hKD3ispAMLSJtJNa%2BOOU5xz7ZdC16p3RUP7mOvInw2JqJ39wPRvo9A5hIA%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}', 'nel': '{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}', 'x-frame-options': 'SAMEORIGIN', 'referrer-policy': 'same-origin', 'cache-control': 'private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0', 'expires': 'Thu, 01 Jan 1970 00:00:01 GMT', 'server': 'cloudflare', 'cf-ray': '8e0d1fc02c0de825-ORD', 'alt-svc': 'h3=\":443\"; ma=86400', 'server-timing': 'cfL4;desc=\"?proto=TCP&rtt=2750&sent=4&recv=6&lost=0&retrans=0&sent_bytes=2830&recv_bytes=824&delivery_rate=1097195&cwnd=249&unsent_bytes=0&cid=4a7e2d545baecbcf&ts=311&x=0\"'} -+-+- b'error code: 502'","time":{"start":1731315962589,"stop":1731315962589,"duration":0}},{"uid":"7a11e28eae349b85","reportUrl":"https://novasamatech.github.io/test-runner/5823//#testresult/7a11e28eae349b85","status":"passed","time":{"start":1731251124969,"stop":1731251125119,"duration":150}},{"uid":"b23f6a79fb073b0a","reportUrl":"https://novasamatech.github.io/test-runner/5822//#testresult/b23f6a79fb073b0a","status":"passed","time":{"start":1731229527340,"stop":1731229527554,"duration":214}},{"uid":"253ff4b1a45b7eb0","reportUrl":"https://novasamatech.github.io/test-runner/5821//#testresult/253ff4b1a45b7eb0","status":"passed","time":{"start":1731164725763,"stop":1731164726042,"duration":279}},{"uid":"a9a84401592a71c2","reportUrl":"https://novasamatech.github.io/test-runner/5820//#testresult/a9a84401592a71c2","status":"passed","time":{"start":1731143130204,"stop":1731143130225,"duration":21}},{"uid":"e446a486d618839b","reportUrl":"https://novasamatech.github.io/test-runner/5819//#testresult/e446a486d618839b","status":"passed","time":{"start":1731078308958,"stop":1731078308981,"duration":23}},{"uid":"e332dbbf4671e8ea","reportUrl":"https://novasamatech.github.io/test-runner/5818//#testresult/e332dbbf4671e8ea","status":"passed","time":{"start":1731056737895,"stop":1731056737980,"duration":85}},{"uid":"709353621fc33996","reportUrl":"https://novasamatech.github.io/test-runner/5817//#testresult/709353621fc33996","status":"passed","time":{"start":1730991924197,"stop":1730991924348,"duration":151}},{"uid":"5402ebdfe969cf19","reportUrl":"https://novasamatech.github.io/test-runner/5816//#testresult/5402ebdfe969cf19","status":"passed","time":{"start":1730970342576,"stop":1730970342598,"duration":22}},{"uid":"4a76d624a082ebfc","reportUrl":"https://novasamatech.github.io/test-runner/5815//#testresult/4a76d624a082ebfc","status":"passed","time":{"start":1730905544141,"stop":1730905544395,"duration":254}},{"uid":"3010853f8e9c3ceb","reportUrl":"https://novasamatech.github.io/test-runner/5814//#testresult/3010853f8e9c3ceb","status":"passed","time":{"start":1730883959451,"stop":1730883959608,"duration":157}},{"uid":"e2b840a9a7d82746","reportUrl":"https://novasamatech.github.io/test-runner/5813//#testresult/e2b840a9a7d82746","status":"passed","time":{"start":1730819153216,"stop":1730819153482,"duration":266}}]},"tags":[]},"source":"236e5a5a5754c13f.json","parameterValues":["{'url': 'wss://khala-api.phala.network/ws', 'name': 'Khala'}"]}